src/github.com/matrix-org/dendrite/appservice/storage/appservice_events_table.go

```diff
@@ -109,59 +100,60 @@ func (s *eventsStatements) selectEventsByApplicationServiceID(
 	limit int,
 ) (
 	txnID, maxID int,
-	events []gomatrixserverlib.ApplicationServiceEvent,
+	events []gomatrixserverlib.Event,
 	err error,
 ) {
 	// Retrieve events from the database. Unsuccessfully sent events first
-	eventRowsCurr, err := s.selectEventsByApplicationServiceIDStmt.QueryContext(ctx, applicationServiceID)
+	eventRows, err := s.selectEventsByApplicationServiceIDStmt.QueryContext(ctx, applicationServiceID)
 	if err != nil {
 		return 0, 0, nil, err
 	}
 	defer func() {
-		err = eventRowsCurr.Close()
+		err = eventRows.Close()
 		if err != nil {
 			log.WithFields(log.Fields{
 				"appservice": applicationServiceID,
 			}).WithError(err).Fatalf("appservice unable to select new events to send")
 		}
 	}()
-	events, maxID, txnID, err = retrieveEvents(eventRowsCurr, limit)
+	events, maxID, txnID, err = retrieveEvents(eventRows, limit)
 	if err != nil {
 		return 0, 0, nil, err
 	}
 
 	return
 }
 
-func retrieveEvents(eventRows *sql.Rows, limit int) (events []gomatrixserverlib.ApplicationServiceEvent, maxID, txnID int, err error) {
+func retrieveEvents(eventRows *sql.Rows, limit int) (events []gomatrixserverlib.Event, maxID, txnID int, err error) {
 	// Get current time for use in calculating event age
 	nowMilli := time.Now().UnixNano() / int64(time.Millisecond)
 
 	// Iterate through each row and store event contents
 	// If txn_id changes dramatically, we've switched from collecting old events to
 	// new ones. Send back those events first.
-	lastTxnID := -2 // Invalid transaction ID
+	invalidTxnID := -2
```

Comment:
this as well can be a const at the top of the file as it does not change and don't have to be recreated on every call.

Reply:
Good point, moved to a global.